lib/csv_sniffer.ex

```diff
@@ -64,24 +64,64 @@ defmodule CsvSniffer do
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?P=delim)'sm,
     #  ".*?",
     ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
     # ,".*?"
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm,
     #  ".*?" (no delim)
-    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm
+    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm
   ]
 
   defp run_quote_regex(sample) do
     Enum.find_value(@quote_regex, {[], []}, fn regex ->
-      case Regex.scan(regex, sample, capture: :all_names) do
-        [] -> false
-        matches -> {Regex.names(regex), matches}
+      names = Regex.names(regex)
+
+      with matches <- Regex.scan(regex, sample, capture: :all_names),
+           [_ | _] = matches <- Enum.filter(matches, &header_compatible?({names, &1}, sample)) do
+        {names, matches}
+      else
+        _ -> false
       end
     end)
   end
 
+  defp header_compatible?({["quote"], _}, _), do: true
+
+  defp header_compatible?({["delim", "quote"], [delim_char, quote_char]}, sample) do
+    quote_regex = ~r/^[^\n#{quote_char}]*#{quote_char}/
+
+    case Regex.match?(quote_regex, sample) do
+      false ->
+        sample
+        |> String.split("\n", parts: 2)
+        |> List.first()
+        |> get_best_delim_candidate()
+        |> Kernel.==(delim_char)
+
+      true ->
```

Comment:
if there is quoting in the header line, we don't complicate ourselves and mark it as "compatible", because that's a very unusual corner case